### PR TITLE
project: block overlay usage with "bare" bases

### PR DESCRIPTION
**Pull request by tigarmo**
- [ ] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?

-----

### CODE REVIEW

File: requirements-dev.txt [hunk 1]
```diff
@@ -18,7 +18,8 @@ commonmark==0.9.1
 coverage==6.5.0
 craft-archives==1.1.0
 craft-cli==2.0.0
-craft-parts==1.21.0
+#craft-parts==1.21.0
+git+https://github.com/canonical/craft-parts.git@CRAFT-1770-has-overlay#egg=craft_parts
```
Comment (tigarmo) [inline requirements-dev.txt:22]: noting this to myself: update before merging

**Review by cjdcordeiro — APPROVED** [review]
Thanks. This LGTM, although it feels like it's patching a bigger and unrelated issue -> `overlay` has, IMO, the connotation that something is running on top of **what you have primed so far**. I.e., if we eventually introduce the concept of ordered parts and OCI layering based on parts, I'd expect the overlay script of layer 5 to run on whatever is already present in layer 4 and below. This is not possible today though, so the only reason I'm raising this point is because I'm concerned that we might be giving the wrong meaning to `overlay`

**Comment by sergiusens** [comment]
Closes #284

File: requirements-dev.txt (hunk 1 above)
Comment (cmatsuoka) [reply requirements-dev.txt:22]: Craft-parts 1.22.0 released.

**Comment by cmatsuoka** [comment]
> I'd expect the overlay script of layer 5 to run on whatever is already present in layer 4 and below.

The layer stack should be assembled up to the layer you're running the overlay script on -- is something different happening?

**Pull request merged**